app/src/androidTest/java/org/resolvetosavelives/red/newentry/search/PatientRepositoryTest.kt

```diff
@@ -0,0 +1,92 @@
+import org.threeten.bp.LocalDate
+
+@RunWith(AndroidJUnit4::class)
+class PatientRepositoryTest {
+
+  lateinit var patientRepository: PatientRepository
+  lateinit var database: AppDatabase
+
+  @Before
+  fun setUp() {
+    val context = InstrumentationRegistry.getTargetContext()
+    database = Room.inMemoryDatabaseBuilder(context, AppDatabase::class.java).build()
+    patientRepository = PatientRepository(database)
+  }
+
+  @Test
+  fun createAnOngoingPatientEntry_thenSaveItCompletely() {
+    val ongoingAddress = OngoingPatientEntry.Address("Noida", "Gautam Buddha Nagar", "Uttar Pradesh")
+    val ongoingPersonalDetails = OngoingPatientEntry.PersonalDetails("Ashok Kumar", "09-10-1942", null, Gender.MALE)
+
+    val personalDetailsOnlyEntry = OngoingPatientEntry(personalDetails = ongoingPersonalDetails)
+
+    patientRepository.saveOngoingEntry(personalDetailsOnlyEntry)
+        .andThen(patientRepository.ongoingEntry())
+        .map { ongoingEntry -> ongoingEntry.copy(address = ongoingAddress) }
+        .flatMapCompletable { withAddress -> patientRepository.saveOngoingEntry(withAddress) }
+        .andThen(patientRepository.saveOngoingEntryAsPatient())
+        .subscribe()
+
+    val search1 = patientRepository.searchPatients("lakshman")
+        .take(1)
+        .blockingFirst()
+
+    assertThat(search1).isEmpty()
+
+    val search2 = patientRepository.searchPatients("ashok")
+        .take(1)
+        .blockingFirst()
+
+    assertThat(search2).isNotEmpty()
+  }
+
+  @Test
+  fun patientWithAddressSearch_shouldReturn_correctlyCombinedObject() {
+    val ongoingAddress = OngoingPatientEntry.Address("Arambol", "Arambol", "Goa")
+    val ongoingPersonalDetails = OngoingPatientEntry.PersonalDetails("Asha Kumar", "15-08-1947", null, Gender.FEMALE)
+
+    val personalDetailsOnlyEntry = OngoingPatientEntry(personalDetails = ongoingPersonalDetails)
+
+    patientRepository.saveOngoingEntry(personalDetailsOnlyEntry)
+        .andThen(patientRepository.ongoingEntry())
+        .map { ongoingEntry -> ongoingEntry.copy(address = ongoingAddress) }
+        .flatMapCompletable { withAddress -> patientRepository.saveOngoingEntry(withAddress) }
+        .andThen(patientRepository.saveOngoingEntryAsPatient())
+        .subscribe()
+
+    val results = patientRepository.searchPatientsWithAddresses("kumar")
+        .take(1)
+        .blockingFirst()
+
+    val patientWithAddress = results[0]
+
+    assertThat(patientWithAddress.fullName).isEqualTo("Asha Kumar")
+    assertThat(patientWithAddress.gender).isEqualTo(Gender.FEMALE)
+    assertThat(patientWithAddress.dateOfBirth).isEqualTo(LocalDate.parse("1947-08-15"))
+    assertThat(patientWithAddress.createdAt).isGreaterThan(patientWithAddress.address.createdAt)
+    assertThat(patientWithAddress.address.colonyOrVillage).isEqualTo("Arambol")
+    assertThat(patientWithAddress.address.state).isEqualTo("Goa")
+    assertThat(patientWithAddress.address.syncPending).isTrue()
```

Comment:
Will it be easier if we create a `PatientWithAddress` object with expected values and compare it with the received value? Instead of comparing each field one by one. 

This way you'll also ensure that any new field to `PatientWithAddress` gets checked.